File: Bowdoin Dining/SettingsViewController.m
//
//  SettingsViewController.m
//  Bowdoin Dining
//
//  Created by Ruben on 7/13/14.
//
//

#import <Foundation/Foundation.h>
#import "AppDelegate.h"
#import "SettingsViewController.h"

@interface SettingsViewController ()
@end

@implementation SettingsViewController
AppDelegate *delegate;

- (void)viewDidLoad {
    [super viewDidLoad];
    
    delegate  = (AppDelegate *)[[UIApplication sharedApplication] delegate];
}

- (void)viewWillAppear:(BOOL)animated {
    //hide the status bar
    [[UIApplication sharedApplication] setStatusBarHidden:NO];
    //if a filter was set, set it again in the view
    [self.dietFilter setSelectedSegmentIndex: [[NSUserDefaults standardUserDefaults] integerForKey:@"diet-filter"]];
    
    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
    NSString *username = [userDefaults objectForKey:@"bowdoin_username"];
    NSString *password = [userDefaults objectForKey:@"bowdoin_password"];
    if(!username || !password) {
        [self.logoutButton setEnabled: FALSE];
        [self.logoutButton setBackgroundColor:[UIColor lightGrayColor]];
    }
}

//user selected a filter (or turnd off)
- (IBAction)indexDidChangeForSegmentedControl: (UISegmentedControl *) sender {
    //tell delegate to update diet filter
    [delegate updateDietFilter:sender.selectedSegmentIndex];
    
    //save to preferences
    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
    [userDefaults setInteger:self.dietFilter.selectedSegmentIndex forKey:@"diet-filter"];
    [userDefaults synchronize];
}

- (IBAction)logout:(UIButton *)sender {
    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
    [userDefaults removeObjectForKey: @"bowdoin_username"];
    [userDefaults removeObjectForKey: @"bowdoin_password"];
    [userDefaults synchronize];
    [self.logoutButton setEnabled: FALSE];
    [self.logoutButton setBackgroundColor:[UIColor lightGrayColor]];
}
@end

File: Bowdoin Dining/SettingsViewController.swift
//
//  SettingsViewController.swift
//  Bowdoin Dining
//
//  Created by Ruben Martinez Jr on 7/17/14.
//  Copyright (c) 2014 Ruben Martinez Jr. All rights reserved.
//

import UIKit

class SettingsViewController: UIViewController {
    var delegate = UIApplication.shared.delegate as! AppDelegate
    @IBOutlet var dietFilter : UISegmentedControl!
    @IBOutlet var logoutButton : UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        //defaults to 0 (off) if none set
        self.dietFilter.selectedSegmentIndex = UserDefaults.standard.integer(forKey: "diet-filter")
        
        if (self.delegate.user != nil && self.delegate.user!.loggedIn) || User.credentialsStored() {
            self.logoutButton.isEnabled = true
            self.logoutButton.backgroundColor = UIColor.red
        } else {
            self.logoutButton.isEnabled = false
            self.logoutButton.backgroundColor = UIColor.lightGray
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override var preferredStatusBarStyle: UIStatusBarStyle {
        return .lightContent
    }

    @IBAction func indexDidChangeForSegmentedControl(_ sender: UISegmentedControl) {
        delegate.updateDietFilter(sender.selectedSegmentIndex)
        
        let userDefaults = UserDefaults.standard
        userDefaults.set(sender.selectedSegmentIndex, forKey: "diet-filter")
        userDefaults.synchronize()
    }
    
    @IBAction func logout(_ sender: AnyObject) {
        if self.delegate.user != nil {
            self.delegate.user!.logout()
        } else {
            User.forget()
        }
        
        self.logoutButton.isEnabled = false
        self.logoutButton.backgroundColor = UIColor.lightGray
    }
}

